src/main/frontend/utils.js

```diff
@@ -136,15 +105,28 @@ export const verifyPermission = async (handle, readWrite) => {
 
 // NOTE: Need externs to prevent `options.recursive` been munged
 //       When building with release.
-export const openDirectory = async (options = {}, cb) => {
+export const openDirectory = async (options = {}) => {
   options.recursive = options.recursive || false;
-  const handle = await window.showDirectoryPicker({
-    mode: 'readwrite'
-  });
-  const _ask = await verifyPermission(handle, true);
-  return [handle, getFiles(handle, options.recursive, cb)];
+  options.startIn = options.startIn || 'documents';
+  options.mode = options.mode || 'readwrite';
+  options.id = options.id || 'open-directory';
+  try {
+    const files = await directoryOpen(options);
+    let handle;
+    if (files.length && !(files[0] instanceof File)) {
+      handle = files[0];
+    } else {
+     handle = files[0].directoryHandle;
```

Comment:
After looking into the implementation, this should be a buggy design decision.
The directoryHandle only binds to the containing directory, not the top directory.

Reply:
Good this got caught.